src/app/tests/TestCommandInteraction.cpp

```diff
@@ -757,25 +741,15 @@ void TestCommandInteraction::TestCommandHandlerCommandEncodeFailure(nlTestSuite
     auto path               = MakeTestCommandPath();
     auto requestCommandPath = ConcreteCommandPath(path.mEndpointId, path.mClusterId, path.mCommandId);
     CommandHandlerWithOutstandingCommand commandHandler(nullptr, requestCommandPath, NullOptional);
-    System::PacketBufferHandle commandPacket;
 
     TestExchangeDelegate delegate;
     auto exchange = ctx.NewExchangeToAlice(&delegate, false);
-    commandHandler.mExchangeCtx.Grab(exchange);
+    commandHandler.mDispatcher.SetExchangeContext(exchange);
```

Comment:
Why is it needed to reach-out this deep? Is there not a more behavioral way to test? This is 2 levels of friendship deep, and  it tests impl more than behavior.

Reply:
This was part of my fear with the CommandResponseSender refactor (formerly `CommandHandlerDispatcher`), that some of these older tests make a little less sense as `CommandHandler` moves away from sending out the responses itself. I will try to go over these on Monday to see what I can do

> This is 2 levels of friendship deep

I mean in this case it isn't SetExchangeContext, the friend is for line 1759